src/Storages/HDFS/HDFSCommon.h

```diff
@@ -86,7 +86,7 @@ static const String CONFIG_PREFIX;
         return config_stor.emplace_back(std::make_pair(k, v));
     }
 
-    hdfsBuilder * hdfs_builder;
+    hdfsBuilder * hdfs_builder = nullptr;
```

Comment:
AFAICS, the constructor will always initialize this so this is reduntant. OTOH, if it fails it's set to nullptr (no exception), so it seems like the fix should be in the destructor, right?

It should be something like:
```
~HDFSBuilderWrapper() { if (hdfs_builder) hdfsFreeBuilder(hdfs_builder); }
```

Otherwise `hdfsFreeBuilder` (which is just delete) is being called in nullptr

Reply:
Yes, but there is also a move constructor `HDFSBuilderWrapper(HDFSBuilderWrapper && other)` which does std::swap of uninitialised `hdfs_builder` with initialized `other.hdfs_builder`. That means that in the destructor of the moved object we will try to perform `hdfsFreeBuilder` and subsequently `delete` on a random address.

As far as I am aware in C++ standard `delete` of nullptr is legal and will not do anything, but if you'd like to be on a safer side I can also add it.